pkg/units/metrics_collection.go

```diff
@@ -52,6 +53,12 @@ func (mc *MetricsCollection) getProgramsVerified() int {
 	return mc.programsVerified
 }
 
+func (mc *MetricsCollection) getCoverageHistory() map[time.Time]int {
+	mc.metricsLock.Lock()
```

Comment:
Do we need this lock?

Reply:
I think go can panic if it detects a race condition... _I think_. it costs nothing to be careful /shruggie